ls1intum/Artemis
# `Communication`: Hide empty channel sections on the messages tab

### Checklist
#### General


#### Client

### Motivation and Context


The sections for exercise, lecture and exam channels are always shown on the messages tab within the sidebar, even if no channels are listed.

### Description

Empty exercise/lecture/exam channel sections are now hidden. The general channel section is always shown, like before. Additionally, to still be able to join other channels, if the category is hidden, the browse-channel modal contains a collapsable section which shows all channels, instead of only showing general/exercise/lecture/exam channels (see screenshots).

### Steps for Testing

Prerequisites:
- 1 Instructor (optional)
- 1 Student

1. Use existing channels or use the instructor to create at least one lecture, exercise and exam in a course with messaging enabled
2. Navigate as a student to the messages tab of this course and leave/join the lecture/exercise/exam but also general channels to check that the lecture/exercise/exam are hidden if the student has left all lecture/exercise/exam channels
3. The general channel section should always be shown
4. The "browse-channel" option next to each channel section shows both the channels of the corresponding category but also all other channels in the (collapsed) section below

### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage







### Screenshots


#### Sidebar
If a user has not joined any exercise/lecture/exam channel, the corresponding category is hidden (except general):
<img width="220" alt="categories hidden" src="https://github.com/ls1intum/Artemis/assets/44754405/8e1565e2-21b8-488b-a2d7-3d227d3ca48c">
It is not hidden, if a user has joined any of the channels within the category:
<img width="220" alt="categories visible" src="https://github.com/ls1intum/Artemis/assets/44754405/8087889b-3c5b-419b-97b6-051fb459a854">

#### Browse-Channels Modal
Other channels are collapsed when opening the modal:
<img width="811" alt="other channels collapsed" src="https://github.com/ls1intum/Artemis/assets/44754405/4146796a-0fe2-4896-9616-5b54de9d947e">
By clicking on "> Other Channels", the remaining channels are shown:
<img width="811" alt="other channels visible" src="https://github.com/ls1intum/Artemis/assets/44754405/da341ccf-538d-46dd-ba14-a70d4c2ef258">
